### DISCUSSION TITLE
How to use a private go library in dagger module?

### DISCUSSION THREAD

**Discussion by azat-dev** [dagger/dagger#9174] [Help]
Hello! 

### Given:
I am trying to daggerize my application, my dagger module uses a privat repository with go library.
The library installs fine on my host machine

### When I execute:
dagger call handle-gitlab-event --source=.

### Then I receive:

running post-command: go mod tidy                                                                                                                    
go: github.com/my_library_name: invalid version: git ls-remote -q origin in /go/pkg/m            
/xxxxxxxxx: exit status 128:                                                                  
        fatal: could not read Username for 'https://github.com': terminal prompts disabled                                                           
Confirm the import path was entered correctly.                                                                                                       

### PS:
As I understand it executes dagger inside a docker container, so it doesn't have access to git credentials
I've read documentation about private modules https://docs.dagger.io/api/remote-modules/, but I think it will not help because I use a private "go library", I think if I will create a private dagger module the problem will persists because it still will not access to my private "go library"

Thank you!

**Comment by grouville** [comment]
Hey 👋 

You're right we currently don't support this use-case -- There are several ways to fix this issue ; one is for the container to forward the git credential request to our buildkit implementation that forwards it to the host ; the issue is that your edge case is also valid for other private modules for all the other languages

This proposal https://github.com/dagger/dagger/issues/9007 could be one way -- nonetheless, I'll try to come up with an intermediary hack on the meantime 😇

**Reply by marcosnils** [reply]
> This proposal #9007 could be one way -- nonetheless, I'll try to come up with an intermediary hack on the meantime 😇

relaying Justin's comment's (https://discord.com/channels/707636530424053791/1318581231465533450) here: 

> hm, i actually don't think that 9007 is actually the solution here
> we'd implement 8809 in a similar way to how the git socket forwarding works i think
> essentially, having a buildkit session attachable of some variety that can load git information from the client
> one of the methods on that would fetch the gitconfig

cc @rajatjindal @grouville

**Comment by azat-dev** [comment]
Hi! 👋

At the moment, I've included my dependencies as a git subtree. 

Thank you for your work. I will wait for next versions 😁
